src/datacontainer.js

```diff
@@ -108,27 +115,137 @@ class DataContainer {
+    } catch (error) {
+      // Ignore the 'AuthorizationPermissionMismatch' error that will be throw if the client has read-only rights.
+      // The save of the schema can be done only by the clients with read-write access.
+      if (error.code !== 'AuthorizationPermissionMismatch') {
+        rethrowDebug(`Failed to save the json schema '${this.schema.id}' with error: ${error}`, error);
+      }
+    }
+  }
+
+  /**
+   * If the schema was previously saved, this method will make an integrity check, otherwise will save the schema in
+   * a blockBlob.
+   * @private
+   */
+  async _cacheSchema() {
+    let storedSchema;
+    let schemaName = this._getSchemaName(this.schemaVersion);
+    try {
+      let schemaBlob = await this.blobService.getBlob(this.name, schemaName);
+      storedSchema = schemaBlob.content;
+    } catch (error) {
+      if (error.code === 'BlobNotFound') {
+        this._saveSchema();
+
+        this._validateFunctionMap[this.schemaVersion] = this.validator.compile(this.schema);
+        return;
+      }
+      rethrowDebug(`Failed to save the json schema '${this.schema.id}' with error: ${error}`, error);
+    }
+
+    // integrity check
+    if (storedSchema !== JSON.stringify(this.schema)) {
+      throw new SchemaIntegrityCheckError('The stored schema is not the same with the schema defined.');
+    }
+    this._validateFunctionMap[this.schemaVersion] = this.validator.compile(this.schema);
+  }
+
+  /**
+   * Method that validates the content
+   *
+   * @param content - JSON content
+   * @param schemaVersion - the schema version (optional)
+   *
+   * @return {object}
+   * ```js
+   * {
+   *    valid: boolean,   // true/false if the content is valid or not
+   *    errors: [],       // if the content is invalid, errors will contain an array of validation errors
+   * }
+   * ```
+   */
+  async validate(content, schemaVersion = this.schemaVersion) {
+    let ajvValidate = this._validateFunctionMap[schemaVersion];
+    // if the validate function is not available, this means that the schema is not yet loaded
+    if (!ajvValidate) {
+      // load the schema
+      try {
+        let schemaBlob = await this.blobService.getBlob(this.name, this._getSchemaName(schemaVersion));
+        let schema = JSON.parse(schemaBlob.content);
+        // cache the ajv validate function
+        this._validateFunctionMap[schemaVersion] = this.validator.compile(schema);
+        ajvValidate = this._validateFunctionMap[schemaVersion];
+      } catch (error) {
+        rethrowDebug(`Failed to save the json schema '${this.schema.id}' with error: ${error}`, error);
+      }
+    }
+    let result = {
+      valid: ajvValidate(content),
+      errors: ajvValidate.errors,
+    };
+
+    return result;
+  }
+
+  async init() {
+    // ensure the existence of the data container
+    await this.ensureContainer();
+
+    // cache the JSON schema
+    await this._cacheSchema();
```

Comment:
When a datacontainer is created, the schema version is optional and the default value is 1, which is set in the constructor. This value is used to load the schema and to check the intergrity (if the schema was previously saved) or to save it. 
this._validateFunctionMap is used to cache the validation function which is generated when the ajv compiles the schema. I created this map, because there are cases when a dataContainer stores blobs with content with different schema version. This means that, when, for example, I modify a blob which has the content created with a schema different than the default one, I need to load the corresponding schema, to compile it and to generate the validation function. All of these are done lazy (on demand) and I cache the validation function in order to be reused for other blobs that will be modified and are using the same schema version.
I changed the method _cacheSchema() a little bit. Now it does only schema caching, instead of schema caching and the generation of validation function. The logic of schema compile is only in validate() method.